### PR TITLE
Add lldb

### PR DESCRIPTION
**Pull request by xoviat**
Closes gh-4756.

### CODE REVIEW

**Comment by isuruf** [comment]
Can you rebase on top of master?

**Comment by xoviat** [comment]
Will do after it works.

File: recipes/lldb/bulid.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+mkdir build
```
Comment (isuruf) [inline recipes/lldb/bulid.sh]: `bulid.sh` -> `build.sh`

File: recipes/lldb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - libxml2 2.9.*
+    - swig
+    - libedit  # [linux]
+
+  run:
+    - vs2015_runtime  # [win]
+
+    - llvmdev =={{ version }}
+    - clangdev =={{ version }}
+    - libedit  # [linux]
```
Comment (jakirkham) [inline recipes/lldb/meta.yaml]: Why is this needed? 😕

Comment (xoviat) [reply recipes/lldb/meta.yaml]: Dynamic loader failed to resolve a symbol at runtim.e

**Comment by xoviat** [comment]
@jakirkham Ideas for resolving the `regex_impl.h` issue? Here it is: https://github.com/llvm-mirror/llvm/blob/master/lib/Support/regex_impl.h

File: recipes/lldb/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipes/lldb/meta.yaml]: What symbol? Do you have a log?

Comment (isuruf) [reply recipes/lldb/meta.yaml]: libedit is a requirement for linux. See https://lldb.llvm.org/build.html

**Comment by xoviat** [comment]
Note: this appears ready to merge.

**Comment by xoviat** [comment]
@isuruf I've confirmed that OSX builds; can you merge?

**Comment by xoviat** [comment]
Note: the CI failures that have happened are due to download failures that have occured and not because the recipe is incorrect.

File: recipes/lldb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+about:
+  home: http://llvm.org/
+  license: NCSA
+  license_file: LICENSE.TXT
+  summary: The LLVM Debugger
+
+extra:
+  recipe-maintainers:
+    - inducer
```
Comment (isuruf) [inline recipes/lldb/meta.yaml]: @inducer, are you okay with being a maintainer here?

Comment (inducer) [reply recipes/lldb/meta.yaml]: I guess.
